File: src/commands/server_side/SumOfAnnualTurnoverCommand.java
package commands.server_side;

import OrganizationDatabase.OrganizationManagerInterface;
import lib.ExecutionStatus;
import network.client.udp.User;

public class SumOfAnnualTurnoverCommand extends ServerSideCommand {

    public SumOfAnnualTurnoverCommand(ServerCallbackFunction callback) {
        super(callback);
    }

    @Override
    protected void executeImplementation(
            User user,
            OrganizationManagerInterface organizationManager,
            Object[] args,
            ServerCallbackFunction callback
    ) throws Exception {
        assert args.length == 0;

        callback.invoke(
                user,
                organizationManager,
                ExecutionStatus.SUCCESS,
                null,
                organizationManager.getSumOfAnnualTurnover()
        );
    }
}

File: src/commands/server_side/SumOfAnnualTurnoverCommand.kt
package commands.server_side

import database.OrganizationManagerInterface
import network.client.udp.User

class SumOfAnnualTurnoverCommand : ServerSideCommand() {
    override suspend fun executeImplementation(
        user: User?,
        organizationManager: OrganizationManagerInterface,
        argument: Any?
    ): Result<Double> {
        assert(argument == null)
        return Result.success(organizationManager.sumOfAnnualTurnover)
    }
}
